### PR TITLE
l3_interfaces unit testing code added

### PR DESCRIPTION
**Pull request by kannankvs**
##### SUMMARY
Unit testing code for l3_interfaces module added.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
l3_interfaces Unit Testing

### CODE REVIEW

**Comment by kannankvs** [comment]
NOTE: UT cases have got a dependency on the [Issue170](https://github.com/ansible-collections/dellemc.enterprise_sonic/issues/170). This will succeed once if the bug is fixed.

**Comment by kannankvs** [comment]
NOTE2: 'replaced' and 'overridden' features are getting modified in l3_interfaces module by developers in parallel. This UT PR can be merged once if the PR for l3_interfaces module for 'replaced' and 'overridden' support is merged. If this UT requires rework after merging the l3_interfaces changes, it will be taken up after that gets merged.

File: plugins/modules/sonic_l3_interfaces.py
Comment (kerry-meyer) [inline plugins/modules/sonic_l3_interfaces.py:1]: Although the change set looks fine, the current l3_interfaces implementation for "replaced" and "overridden" states is incorrect.

I would like to hold off on documenting support for these states in this resource module until after we have completed the work to implement these states correctly.

**Review by kerry-meyer — COMMENTED** [review]
Although the change set looks fine, the current l3_interfaces implementation for "replaced" and "overridden" states is incorrect.

I would like to hold off on documenting support for these states in this resource module until after we have completed the work to implement these states correctly.

File: plugins/module_utils/network/sonic/argspec/l3_interfaces/l3_interfaces.py
Comment (kerry-meyer) [inline plugins/module_utils/network/sonic/argspec/l3_interfaces/l3_interfaces.py:1]: Although the argspec change set looks fine, the current l3_interfaces implementation for "replaced" and "overridden" states is incorrect.

I would like to hold off on documenting support for these states in this resource module until after we have completed the work to implement these states correctly.

For now, it would be best to delete the changes for this file so we can merge the UT portion of the PR change set when it is working correctly.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Please delete the premature, unrelated changes in the argspec and module files in order to proceed with this PR after the blocking bug in the l3_interfaces resource module is fixed.

**Comment by kerry-meyer** [comment]
> The code coverage sanity test is failing for this PR. The failures are related to problems in this change set.
> 
> Please fix the failures and delete the premature, unrelated changes in the argspec and module files in order to proceed with this PR.

I noticed your note and realize now that the sanity failures are not caused by the proposed UT changes. I understand that we will need to wait for the fix to https://github.com/ansible-collections/dellemc.enterprise_sonic/issues/170 before proceeding with this PR. 

If the l3_interfaces bug is fixed before correct support is added for "replaced" and "overridden" states, it would still be good to defer the argspec and module documentation changes to the PR that provides the real l3_interfaces support for these states.

**Comment by kannankvs** [comment]
Changes related to "replaced" and "overridden" in plugins directory are not required and hence removed them now. They should be synced from the PR where the actual code for supporting "replaced" and "overridden" are implemented.

**Comment by kannankvs** [comment]
@kerry-meyer : I removed code for testing replaced and overridden. When I remove UT for replaced and overridden, the coverage percentage decreases below 70. Hence, I added few new combinations and improved the coverage above 70% now. Please review and merge.

**Review by kerry-meyer — APPROVED** [review]

**Pull request merged**
